app/controllers/experiments_controller.rb

```diff
@@ -96,6 +96,7 @@ def table
 
     @project = @experiment.project
     @active_modules = @experiment.my_modules.active.order(:name)
+    @my_module_visible_table_columns = current_user.settings['visible_my_module_table_columns'].presence || []
```

Comment:
You meed to define default visible state, not empty array

Reply:
it has to be empty if all the columns are hidden, so the logic afterwards will hide all the columns in the backend too, only visible columns are added inside that array, if all the presence is nil, it just means that every column is hidden, not shown

Reply:
So when the user first time comes to the table, will all columns be hidden?

Reply:
aha, i just checked and you're right, all columns are hidden, i will fix it